src/modules/logger/logger.cpp

```diff
@@ -708,14 +671,58 @@ void Logger::run()
 			/* wait for lock on log buffer */
 			_writer.lock();
 
+			const ssize_t buffer_size = _writer.get_buffer_size_file(LogType::Full);
+
+			// only check buffer if not using mavlink backend
+			const bool check_buffer_fill = (buffer_size > 0) && !_writer.is_started(LogType::Full, LogWriter::BackendMavlink);
+
 			for (int sub_idx = 0; sub_idx < _num_subscriptions; ++sub_idx) {
 				LoggerSubscription &sub = _subscriptions[sub_idx];
 				/* if this topic has been updated, copy the new data into the message buffer
 				 * and write a message to the log
 				 */
-				const bool try_to_subscribe = (sub_idx == next_subscribe_topic_index);
+				bool msg_copied = false;
```

Comment:
I really want to avoid spaghetti code - can you keep the separate method?

Reply:
Then we have to pass in the buffer size and the flag to check it or not. I could go either way.

Reply:
Yes, it's inlined.